modules/mysql/collect_queries_statistics.go

```diff
@@ -0,0 +1,52 @@
+package mysql
+
+import (
+	"time"
+)
+
+// Table Schema:
+// (MariaDB) https://mariadb.com/kb/en/information-schema-processlist-table/
+// (MySql) https://dev.mysql.com/doc/refman/5.7/en/information-schema-processlist-table.html
+const (
+	queryInfoSchemaProcessList = `SELECT TIME,USER FROM INFORMATION_SCHEMA.PROCESSLIST 
+WHERE Info IS NOT NULL AND Info NOT LIKE '%PROCESSLIST%' 
+ORDER BY TIME`
+)
+
+func (m *MySQL) collectProcessListStatistics(collected map[string]int64) error {
+	m.Debugf("executing query: '%s'", queryInfoSchemaProcessList)
+
+	var (
+		queryDuration              int64 // queryInfoSchemaProcessList execution time (not including row fetching)
+		longestRunningQuerySeconds int64 // slowest query milliseconds in process list
+		user                       string
+	)
+
+	start := time.Now()
+	rows, err := m.db.Query(queryInfoSchemaProcessList)
+	queryDuration = int64(time.Since(start).Seconds())
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	collected["process_list_query_count_system"] = 0
+	collected["process_list_query_user"] = 0
+
+	for rows.Next() {
+		if err := rows.Scan(&longestRunningQuerySeconds, &user); err != nil {
+			return err
+		}
```

Comment:
We should do it only for the first row

Reply:
In our query we order by time `ASC`. The last entry low will also be the maximum time. We could also scan the opposite way and scan the first row only. Not sure what's best.

Reply:
You are right! We are good if we use `ASC`. Is `ASC` the default one? I see we don't use it explicitly in [the query](https://github.com/georgeok/go.d.plugin/blob/9e7d580dee3f8bfb4398be8d3ae268306f3a8dff/modules/mysql/collect_queries_statistics.go#L11-L13).

Reply:
Yes `ASC` is the default. If I use it explicitly I get a lint warning!

Reply:
Got it, thanks!